hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -827,7 +828,7 @@ public AppsInfo getApps(HttpServletRequest hsr, String stateQuery,
     });
 
     if (apps.getApps().isEmpty()) {
-      return null;
+      return new AppsInfo();
```

Comment:
After comparing the results from the ResourceManager (RM), if it doesn't return any results, we should return an empty object instead of null.